src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseDefaultDllImportSearchPathsAttribute.cs

```diff
@@ -62,10 +62,10 @@ public override void Initialize(AnalysisContext context)
             {
                 var compilation = compilationStartAnalysisContext.Compilation;
                 var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilation);
-
-                if (!wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDllImportAttribute, out INamedTypeSymbol? dllImportAttributeTypeSymbol) ||
-                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDefaultDllImportSearchPathsAttribute, out INamedTypeSymbol? defaultDllImportSearchPathsAttributeTypeSymbol) ||
-                    compilationStartAnalysisContext.Compilation.SyntaxTrees.FirstOrDefault() is not SyntaxTree tree)
+                if ((!wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDllImportAttribute, out INamedTypeSymbol? dllImportAttributeTypeSymbol)
+                        & !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesLibraryImportAttribute, out INamedTypeSymbol? libraryImportAttributeTypeSymbol))
```

Comment:
Is this supported to be `&&`?

Reply:
To make sure the `out` variable is initialized, it needs to be a non-short-circuiting `&`

Reply:
Can we split the check into multiple steps instead? For example:
```c#
bool hasDllImportAttr = wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(...);
bool hasLibraryImportAttr = wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(...);
if ((!hasDllImportAttr && !hasLibraryImportAttr)
    || ...)
```

I think having non-obvious requirements and side effects like that wrapped up in this if condition makes the code harder to understand and maintain.